pkg/template/template.go

```diff
@@ -839,3 +840,19 @@ func AsMapOfStrings(context string, yml string, values map[string]interface{}) (
 
 	return obj, nil
 }
+
+// splitListTrim trims the separator from prefix and suffix, then
+// returns an array of strings split on separator
+//
+// NOTE:
+//  This is intended to be used as a template function
+//
+// Example:
+//  {{- $pools =: "pool1;pool2;" | splitListTrim ";" | -}}
+//
+// Above operation would assign an array of strings containing
+// 'pool1' and 'pool2' to pools
+func splitListTrim(sep, orig string) []string {
+	processedStr := strings.TrimSuffix(strings.TrimPrefix(orig, sep), sep)
```

Comment:
does TrimRight and TrimLeft makes sense?

Reply:
Could create it but I don't see any utility as of now. The only edge case it the right part.

Reply:
https://golang.org/pkg/strings/#TrimRight and similarly, TrimLeft